beginner.md

```diff
@@ -0,0 +1,588 @@
+to install the required dependencies and create a basic scaffold.
+
+Install the `generator-delite-element` globally (or update it if necessary).
+
+    npm install -g generator-delite-element
+
+##Templates
+`delite` provides first class support for templates. We wouldn't expect to programmatically create DOM nodes & this is where
+`delite` comes into it's own; out of the box `delite` supports templating using a built in implementation of [Handlebars](http://handlebarsjs.com/).
+
+Note there are some limitations using the `delite/handlebars!` plugin for templating, namely it doesn't support iterators or conditionals.
+However in many cases this isn't a limiting factor. Support for this will be explained in a later more advanced tutorial when we discuss
+[Liaison](https://github.com/ibm-js/liaison). The handlebars template implementation `delite` uses is primarily focused on performance.
+
+Let's try create a 'real life' widget, for example a blogging widget.
+
+### create the scaffold
+We'll create a new `delite` custom element using Yeoman.
+
+From the command line create a new directory somewhere (named `blogging-package`) and change directory to it using the commands:
+
+    mkdir -p blogging-package
+    cd blogging-package
+
+Run Yeoman to create our scaffold.
+
+    yo delite-element
+
+You'll be prompted to enter the widget package name and the name of the custom widget element. Set the following choices shown in brackets below.
+
+    ? What is the name of your delite widget element package? (blogging-package)
+    ? What do you want to call your delite widget element (must contain a dash)? (blog-post)
+    ? Would you like your delite element to be built on a template? (Y)
+    ? Would you like your delite element to providing theming capabilities? (N)
+    ? Will your delite element require string internationalization? (N)
+    ? Will your delite element require pointer management? (N)
+    ? Do you want to use build version of delite package (instead of source version)? (N)
+
+### What's been generated
+Yeoman created the following (as shown in the console output):
+
+- `./BlogPost.js` - __this is our widget module__
+- `./BlogPost/css/BlogPost.css` - __this is our widget css__
+- `./BlogPost/BlogPost.html` - __this is our widget template__
+- `./samples/BlogPost.html` - __this is a sample how to use our new widget__
+
+You can view the sample generated HTML `./samples/BlogPost.html` in a browser to see what's been created.
+
+
+## Creating a custom element
+Viewing the `./samples/BlogPost.html` example HTML we can see we've (partly) created the custom element declaratively in markup via:
+```html
+<blog-post id="element" value="The Title"></blog-post>
+```
+For those who used the Dojo Toolkit Dijit framework previously, an important conceptual difference in `delite` is that the widget is the DOM node.
```

Comment:
Same here, this is really of interest only to Dijit people. Maybe said as a footnote or something like that to not break the reading flow of others.